include/ensmallen_bits/qhadam/qhadam.hpp

```diff
@@ -0,0 +1,164 @@
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef ENSMALLEN_ADAM_QHADAM_HPP
+#define ENSMALLEN_ADAM_QHADAM_HPP
+
+ #include <ensmallen_bits/sgd/sgd.hpp>
+#include "qhadam_update.hpp"
+
+ namespace ens {
+
+ /**
+ * QHadam is an variation of Adam with Quasi-Hyperbolic step. It can be
+ * a weighted mean of the momentum step. Due to its paramterisation it can
+ * recover many other optimisation strategies.
+ *
+ * For more information, see the following.
+ *
+ * @code
+ * @inproceedings{ma2019qh,
+ *   title={Quasi-hyperbolic momentum and Adam for deep learning},
+ *   author={Jerry Ma and Denis Yarats},
+ *   booktitle={International Conference on Learning Representations},
+ *   year={2019}
+ * }
+ * @endcode
+ *
+ * QHAdam can optimize differentiable separable functions.
+ * For more details, see the documentation on function
+ * types included with this distribution or on the ensmallen website.
+ */
+class QHAdam
+{
+ public:
+  /**
+   * Construct the QHAdam optimizer with the given function and parameters.
+   * QHAdam is sensitive to its paramters and hence a good hyper paramater
+   * selection is necessary as its default may not fit every case.
+   *
+   * The maximum number of iterations refers to the maximum number of
+   * points that are processed (i.e., one iteration equals one point; one
+   * iteration does not equal one pass over the dataset).
+   *
+   * @param stepSize Step size for each iteration.
+   * @param batchSize Number of points to process in a single step.
+   * @param v1 The first quasi-hyperbolic term.
+   * @param v1 The second quasi-hyperbolic term.
+   * @param beta1 Exponential decay rate for the first moment estimates.
+   * @param beta2 Exponential decay rate for the weighted infinity norm
+            estimates.
+   * @param epsilon Value used to initialise the mean squared gradient parameter.
+   * @param maxIterations Maximum number of iterations allowed (0 means no
+   *        limit).
+   * @param tolerance Maximum absolute tolerance to terminate algorithm.
+   * @param shuffle If true, the function order is shuffled; otherwise, each
+   *        function is visited in linear order.
+   * @param resetPolicy If true, parameters are reset before every Optimize
+   *        call; otherwise, their values are retained.
+   */
+   QHAdam(const double stepSize = 0.001,
+          const size_t batchSize = 32,
+          const double v1 = 0.7,
+          const double v2 = 1,
+          const double beta1 = 0.9,
+          const double beta2 = 0.999,
+          const double epsilon = 1e-8,
+          const size_t maxIterations = 100000,
+          const double tolerance = 1e-5,
+          const bool shuffle = true,
+          const bool resetPolicy = true);
+
+   /**
+   * Optimize the given function using QHAdam. The given starting point will be
```

Comment:
The spacing seems off on this comment, it should be of the form

```
/**
 * comment
 */
```

(note the alignment of the `*`)